### PR TITLE
Use iso_c types for fortran to ensure compatibility for the cwrapper

### PR DESCRIPTION
**Pull request by mostafaMamoni**
- [x] Use iso_c_binding to specify precision

- [x] Mark all structures and functions in the f2py_X.F90 file with bind (c)

### CODE REVIEW

File: pyccel/codegen/printing/fcode.py [hunk 1]
```diff
@@ -269,14 +270,14 @@ def _handle_fortran_specific_a_prioris(self, var_list):
...
         self._handle_fortran_specific_a_prioris(self.parser.get_variables(self._namespace))
-
         name = self._print(expr.name)
         name = name.replace('.', '_')
         if not name.startswith('mod_') and self.prefix_module:
             name = '{prefix}_{name}'.format(prefix=self.prefix_module,
                                             name=name)
 
         imports = ''.join(self._print(i) for i in expr.imports)
+        imports += 'use ISO_C_BINDING'
```
Comment (EmilyBourne) [inline pyccel/codegen/printing/fcode.py:280]: This should end with a `\n`

File: pyccel/codegen/printing/fcode.py [hunk 2]
```diff
@@ -327,9 +328,9 @@ def _print_Module(self, expr):
 
     def _print_Program(self, expr):
         self._handle_fortran_specific_a_prioris(self.parser.get_variables(self._namespace))
-
         name    = 'prog_{0}'.format(self._print(expr.name)).replace('.', '_')
         imports = ''.join(self._print(i) for i in expr.imports)
+        imports += 'use ISO_C_BINDING'
```
Comment (EmilyBourne) [inline pyccel/codegen/printing/fcode.py:333]: This should end with a `\n`

**Review by EmilyBourne — CHANGES_REQUESTED** [review]
This looks good to me but it seems to be failing for almost every test on travis. Are you sure f2py knows how to handle iso_c_binding?

**Comment by yguclu** [comment]
> This looks good to me but it seems to be failing for almost every test on travis. Are you sure f2py knows how to handle iso_c_binding?

I agree, most likely there is a problem with f2py. @mostafaMamoni Do the tests run on your machine?

**Comment by yguclu** [comment]
If indeed f2py does not understand the `use iso_c_binding` instruction, that would mean that this PR and #527 need each other to work. Hence I suggest that we merge this branch into [`ebourne_remove_f2py`](https://github.com/pyccel/pyccel/tree/ebourne_remove_f2py).

**Comment by EmilyBourne** [comment]
This branch is now merged into PR #527 as the 2 are inter-dependant

**Pull request closed without merging**